### PR TITLE
Add SVG color glyph support

### PR DESCRIPTION
**Pull request by zauguin**
The code is almost a direct copy from the PNG parts.

### CODE REVIEW

**Comment by deepakjois** [comment]
Is it possible to add a test for this in `spec/harfbuzz_spec.lua`. I see that the PNG stuff has a single test. If there is anything you could assert on after calling this function, that should be enough I guess.

File: src/luaharfbuzz/face.c [hunk 1]
```diff
@@ -398,6 +398,31 @@ static int face_ot_color_has_png(lua_State *L) {
...
+  Face *f = (Face *)luaL_checkudata(L, 1, "harfbuzz.Face");
+  hb_codepoint_t gid = (hb_codepoint_t) luaL_checkinteger(L, 2);
+  hb_blob_t* blob = hb_ot_color_glyph_reference_svg(*f, gid);
+
+  if (hb_blob_get_length(blob) != 0) {
+    Blob *b = (Blob *)lua_newuserdata(L, sizeof(*b));
+    luaL_getmetatable(L, "harfbuzz.Blob");
+    lua_setmetatable(L, -2);
+
+    *b = hb_ot_color_glyph_reference_svg(*f, gid);
```
Comment (khaledhosny) [inline src/luaharfbuzz/face.c]: That is probably a thinko in my part in the PNG code, but you can save the extra call to `hb_ot_color_glyph_reference_svg()` and reuse the `blob` variable.
```suggestion
    *b = blob;
```

Comment (zauguin) [reply src/luaharfbuzz/face.c]: @khaledhosny I fully agree, but I thought that if I change it here, then it also should be changed in the PNG version to be consistent and that wouldn't really fit into this PR.

Comment (khaledhosny) [reply src/luaharfbuzz/face.c]: I’d change both in a separate commit.

Comment (zauguin) [reply src/luaharfbuzz/face.c]: @khaledhosny Done

**Comment by khaledhosny** [comment]
I added a test similar to the PNG one.

**Pull request merged**